helical/benchmark/benchmark.py

```diff
@@ -30,10 +32,13 @@ def evaluate_integration(data_list: list[tuple[str, AnnData, str]], cfg: DictCon
 
     """
     evaluations = {}
-    for model, adata, embed_obsm_name in data_list:
+    for model, embed_obsm_name in data_list:
         LOGGER.info(f"Processing integration evaluation using...")
-        evaluation = _get_integration_evaluations(adata,
-                                                  adata,
+
+        # because scib library modifies the adata object, we need to deepcopy it for each model
+        # otherwise, some evaluations will be identical and thus incorrect 
+        evaluation = _get_integration_evaluations(deepcopy(adata),
+                                                  deepcopy(adata),
```

Comment:
More specifically, this happened here on line 200:
![Uploading Screenshot 2024-07-04 at 17.27.45.png…]()